app/src/main/java/at/bitfire/icsdroid/ui/viewmodel/ValidationModel.kt

```diff
@@ -0,0 +1,103 @@
+/***************************************************************************************************
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ **************************************************************************************************/
+
+package at.bitfire.icsdroid.ui.viewmodel
+
+import android.app.Application
+import android.net.Uri
+import android.util.Log
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import at.bitfire.ical4android.Css3Color
+import at.bitfire.ical4android.Event
+import at.bitfire.ical4android.ICalendar
+import at.bitfire.icsdroid.CalendarFetcher
+import at.bitfire.icsdroid.Constants
+import at.bitfire.icsdroid.HttpUtils.toURI
+import at.bitfire.icsdroid.HttpUtils.toUri
+import at.bitfire.icsdroid.ui.ResourceInfo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import net.fortuna.ical4j.model.property.Color
+import okhttp3.MediaType
+import java.io.InputStream
+import java.io.InputStreamReader
+
+class ValidationModel(application: Application): AndroidViewModel(application) {
```

Comment:
Is there a reason for leaving this in the ui package? I think it'd be good to have it in the model package no?

Reply:
Absolutely, I didn't realize there was already a package for viewmodels 😅